Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift

```diff
@@ -0,0 +1,186 @@
+        changedAngleInDegrees = changedAngleInDegrees.truncatingRemainder(dividingBy: 360.0)
+
+        // Constraining `changedAngleInDegrees` to -30.0 to +30.0 degrees
+        if isRotationAllowed() == false && abs(pinchScale) < 10 {
+            changedAngleInDegrees = changedAngleInDegrees < -30.0 ? -30.0 : changedAngleInDegrees
+            changedAngleInDegrees = changedAngleInDegrees > 30.0 ? 30.0 : changedAngleInDegrees
+        }
+
+        cameraManager.setCamera(to: CameraOptions(bearing: CLLocationDirection(changedAngleInDegrees)),
+                                animated: false,
+                                duration: 0,
+                                completion: nil)
+    }
+
+    internal func rotationEnded(with finalAngle: CGFloat, and anchor: CGPoint, with pinchState: UIGestureRecognizer.State) {
+        var finalAngleInDegrees = finalAngle * 180.0 / .pi * -1
+        finalAngleInDegrees = finalAngleInDegrees.truncatingRemainder(dividingBy: 360.0)
+        cameraManager.setCamera(to: CameraOptions(bearing: CLLocationDirection(finalAngleInDegrees)),
+                                animated: false,
+                                duration: 0,
+                                completion: nil)
+    }
+
+    internal func unrotateIfNeededForGesture(with pinchState: UIGestureRecognizer.State) {
+        guard let mapView = cameraManager.mapView else {
+            return
+        }
+
+        // Avoid contention with in-progress gestures
+        // let toleranceForSnappingToNorth: CGFloat = 7.0
+        if mapView.cameraView.bearing != 0.0
+            && pinchState != .began
+            && pinchState != .changed {
+            if mapView.cameraView.bearing != 0.0 && isRotationAllowed() == false {
+                cameraManager.setCamera(to: CameraOptions(bearing: 0),
+                                        animated: false,
+                                        duration: 0,
+                                        completion: nil)
+            }
+
+            // TODO: Add snapping behavior to "north" if bearing is less than some tolerance
+            // else if abs(self.mapView.cameraView.bearing) < toleranceForSnappingToNorth
+            //            || abs(self.mapView.cameraView.bearing) > 360.0 - toleranceForSnappingToNorth {
+            //    self.transitionBearing(to: 0.0, animated: true)
+            //}
+        }
+    }
+
+    internal func initialPitch() -> CGFloat {
+        guard let mapView = cameraManager.mapView else {
+            return 0
+        }
+        return mapView.cameraView.pitch
+    }
+
+    internal func horizontalPitchTiltTolerance() -> Double {
+        return 45.0
+    }
+
+    internal func pitchChanged(newPitch: CGFloat) {
+        cameraManager.setCamera(to: CameraOptions(pitch: newPitch),
+                                animated: false,
+                                duration: 0,
+                                completion: nil)
+    }
+
+    internal func pitchEnded() {
```

Comment:
empty function, let's at least a "not yet implemented" statement or remove it entirely

Reply:
this existed previously. I've just moved this code out of the `GestureManager.swift` file